juniper/manifest.py

```diff
@@ -61,3 +61,17 @@ def all_keys(manifest_definition, key):
 
     result = [function[key] for function in manifest_definition['functions'].values()]
     return result if isinstance(result[0], str) else chain.from_iterable(result)
+
+
+def filter_manifest_definition(manifest_definition, name_filter):
+    """
+    Filters the manifest to only include functions that partially match
+    the specified filter.
+    :param manifest_definition: Dictionary of the manifest
+    :param name_filter: A function name specified in the manifest
+    :return: Filtered manifest definition
+    """
+    manifest_definition['functions'] = {key: value for (key, value)
+                                        in manifest_definition['functions'].items()
+                                        if key.lower().startswith(name_filter)}
```

Comment:
I would flip this statement to be something like `name_filter in key.lower()` that way you can build a single function or multiple functions as long as the name_filter or keyword appears in the name of the function